tests/unit/tf/outputs/test_contrastive.py

```diff
@@ -0,0 +1,238 @@
+
+
+def test_constrastive_mf_weights_in_output(ecommerce_data: Dataset):
+    schema = ecommerce_data.schema
+    schema["item_id"] = schema["item_id"].with_tags([Tags.TARGET])
+    user_id = schema.select_by_tag(Tags.USER_ID)
+    item_id = schema.select_by_tag(Tags.ITEM_ID)
+
+    # TODO: Change this for new RetrievalModel
+    encoder = mm.TabularBlock(mm.EmbeddingTable(64, user_id.first), aggregation="concat")
+
+    mf = mm.Model(encoder, mm.ContrastiveOutput(item_id, "in-batch"))
+
+    testing_utils.model_test(mf, ecommerce_data, run_eagerly=True)
+
+
+def test_two_tower_constrastive(ecommerce_data: Dataset):
+    model = mm.RetrievalModel(
+        mm.TwoTowerBlock(ecommerce_data.schema, query_tower=mm.MLPBlock([8])),
+        mm.ContrastiveOutput(
+            ecommerce_data.schema.select_by_tag(Tags.ITEM_ID),
+            negative_samplers="in-batch",
+            candidate_name="item",
+        ),
+    )
+
+    testing_utils.model_test(model, ecommerce_data)
+
+
+@pytest.mark.parametrize("run_eagerly", [True, False])
+def test_contrastive_output(ecommerce_data: Dataset, run_eagerly):
+    schema = ecommerce_data.schema
+    schema["item_category"] = schema["item_category"].with_tags(
+        schema["item_category"].tags + "target"
+    )
+    ecommerce_data.schema = schema
+    model = mm.Model(
+        mm.InputBlock(schema),
+        mm.MLPBlock([8]),
+        mm.ContrastiveOutput(
+            schema["item_category"],
+            negative_samplers=PopularityBasedSamplerV2(max_id=100, max_num_samples=20),
+        ),
+    )
+
+    _, history = testing_utils.model_test(model, ecommerce_data, run_eagerly=run_eagerly)
+
+    assert set(history.history.keys()) == {
+        "loss",
+        "ndcg_at_10",
+        "precision_at_10",
+        "map_at_10",
+        "mrr_at_10",
+        "recall_at_10",
+        "regularization_loss",
+    }
+
+
+def test_setting_negative_sampling_strategy(sequence_testing_data: Dataset):
+    dataloader, schema = _next_item_loader(sequence_testing_data, to_one_hot=False)
+    model_out = mm.ContrastiveOutput(schema["item_id_seq"], "in-batch")
+    model = mm.Model(mm.InputBlockV2(schema), mm.MLPBlock([32]), model_out)
+    model.compile(optimizer="adam")
+
+    batch = next(iter(dataloader))
+    output = model(batch[0], batch[1], training=True)
+    assert output[1].shape == (batch[1].shape[0], 51)
+
+    model_out.compile(
+        negative_sampling=[PopularityBasedSamplerV2(max_id=51996, max_num_samples=20)],
+    )
+
+    output = model(batch[0], batch[1], training=True)
+    assert output.outputs.shape == (batch[1].shape[0], 21)
+
+    model.model_outputs[0].compile(
+        negative_sampling=["in-batch", PopularityBasedSamplerV2(max_id=51996, max_num_samples=20)],
+    )
+    output = model(batch[0], batch[1], training=True)
+    assert output.outputs.shape == (batch[1].shape[0], 71)
+
+
+# def test_contrastive_dot_product(ecommerce_data: Dataset):
```

Comment:
are these tests intentionally commented out? If so perhaps a comment explaining why may be helpful if this is  intended as a reminder of something?